### PR TITLE
Moment.js to Date-Fns Migration

### PR DESCRIPTION
**Pull request by utku-ozturk**
Trello: https://trello.com/c/eEhhxmcm

Draft release: [0.1.36b6](https://github.com/4dn-dcic/shared-portal-components/releases/tag/0.1.36b6)
FF PR: https://github.com/4dn-dcic/fourfront/pull/1617

Based on benchmarks and features, we decided to give [date-fns](https://github.com/date-fns/date-fns) library a chance.

### Features Comparison Table: 

features

### Parsing Performance

parse

### Get-Set Performance

getset

### CODE REVIEW

File: src/components/util/search-filters.js
Comment (Bianca-Morris) [inline src/components/util/search-filters.js:39]: Is this not actually in use anywhere at the moment? Just double checking it wasn't accidentally left commented out.

**Review by Bianca-Morris — COMMENTED** [review]
Code changes seem good at a glance; I haven't gotten a chance to test this though, so I'll leave that to Alex. 

As a side note: I am curious what you used for benchmarking/profiling of the different potential libraries (I think you may have mentioned that during your presentation; hope you don't mind reminding me).

File: src/components/ui/LocalizedTime.js
Comment (alexkb0009) [inline src/components/ui/LocalizedTime.js:157]: Can we have this auto-determine what locale the browser is in? We don't want to use enUS if user(s) are not from US.

File: src/components/util/search-filters.js
Comment (alexkb0009) [reply src/components/util/search-filters.js:39]: We do have "date_created" field which is meant to be date_histogram, but don't think it's currently used as facet. This feature might not have been fully implemented in which case is worth removing or commenting out, but am not 100% sure. Would probably just comment out entire `if` block though so that it does `throw new Error('Histogram not currently supported.');` (line below) if `facet.aggregation_type === 'date_histogram'`.

Comment (alexkb0009) [reply src/components/util/search-filters.js:39]: Just need to comment out line 34 -> `// } else if (facet.aggregation_type === 'date_histogram'){`

**Review by alexkb0009 — APPROVED** [review]
This should be all set, I just need to update CGAP to use date-fns as well before merging this.

**Pull request merged**
